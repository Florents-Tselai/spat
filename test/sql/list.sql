/* -------------------- LISTS -------------------- */

/* LPUSH -> LEN -> LPOP */
SELECT LPUSH('list1', 'elem1');
SELECT LLEN('list1'); -- 1
SELECT LPUSH('list1', 'elem2');
SELECT LLEN('list1'); -- 2

SELECT SPGET('list1');

SELECT LPOP('list1'); -- elem2
SELECT LLEN('list1'); -- 1

SELECT LPOP('list1'); -- elem1
SELECT LLEN('list1'); -- 0

SELECT LPOP('list1'); -- null
SELECT LLEN('list1'); -- 0

SELECT LPOP('list1'); -- null
SELECT LLEN('list1'); -- 0

SELECT SPGET('list1');

SELECT DEL('list1');

SELECT DEL('list404');
