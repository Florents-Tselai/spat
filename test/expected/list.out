/* -------------------- LISTS -------------------- */
/* LPUSH -> LEN -> LPOP */
SELECT LPUSH('list1', 'elem1');
 lpush 
-------
 
(1 row)

SELECT LLEN('list1'); -- 1
 llen 
------
    1
(1 row)

SELECT LPUSH('list1', 'elem2');
 lpush 
-------
 
(1 row)

SELECT LLEN('list1'); -- 2
 llen 
------
    2
(1 row)

SELECT SPGET('list1');
  spget   
----------
 list (2)
(1 row)

SELECT LPOP('list1'); -- elem2
 lpop  
-------
 elem2
(1 row)

SELECT LLEN('list1'); -- 1
 llen 
------
    1
(1 row)

SELECT LPOP('list1'); -- elem1
 lpop  
-------
 elem1
(1 row)

SELECT LLEN('list1'); -- 0
 llen 
------
    0
(1 row)

SELECT LPOP('list1'); -- null
 lpop 
------
 
(1 row)

SELECT LLEN('list1'); -- 0
 llen 
------
    0
(1 row)

SELECT LPOP('list1'); -- null
 lpop 
------
 
(1 row)

SELECT LLEN('list1'); -- 0
 llen 
------
    0
(1 row)

SELECT SPGET('list1');
  spget   
----------
 list (0)
(1 row)

